library/src/main/kotlin/org/codice/compliance/Common.kt

```diff
@@ -162,19 +164,30 @@ fun Node.prettyPrintXml(): String {
     return output.toString()
 }
 
-@Suppress("TooGenericExceptionCaught")
+//@Suppress("TooGenericExceptionCaught")
+//fun String.prettyPrintXml(): String {
+//    return try {
+//        Common.buildDom(this).prettyPrintXml()
+//    } catch (e: Exception) {
+//        Log.debugWithSupplier { "'$this' is not valid XML." }
+//        this
+//    }
+//}
 fun String.prettyPrintXml(): String {
-    return try {
-        Common.buildDom(this).prettyPrintXml()
-    } catch (e: Exception) {
-        Log.debugWithSupplier { "'$this' is not valid XML." }
-        this
-    }
+    // Escape all ampersands because Keycloak does not properly escape it in POST responses
+    // which causes the transform to fail.
+    val escapedString = this.replace(Regex("&([^;]+(?!(?:\\\\w|;)))"),
+            { match -> "&amp;${match.value.removePrefix("&")}" })
+    val input = StreamSource(StringReader(escapedString))
+    val output = StreamResult(StringWriter())
+    val transformer = createTransformer()
+    transformer.transform(input, output)
+    return output.writer.toString()
```

Comment:
@coyotesqrl Your last commit made it so HTTP response bodies were printed all on one line for some reason. If you know why off the top of your head I can put the fix in this PR. Otherwise, I'm just reverting for String for now so that we can merge this PR.

Reply:
Don't know why it would do that. It definitely should not have. I'll look after this is merged.